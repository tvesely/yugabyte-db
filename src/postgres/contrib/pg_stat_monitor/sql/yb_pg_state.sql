CREATE EXTENSION pg_stat_monitor;
SELECT pg_stat_monitor_reset();
-- https://github.com/yugabyte/yugabyte-db/issues/11801
-- TODO: pg_stat_monitor_reset does not remove this from shared memory and hence this might appear
-- in subsequent pg_stat_monitor tables.
-- SELECT 1;
SELECT 1/0;   -- divide by zero

SELECT query, state_code, state FROM pg_stat_monitor ORDER BY query COLLATE "C";
SELECT pg_stat_monitor_reset();
DROP EXTENSION pg_stat_monitor;
