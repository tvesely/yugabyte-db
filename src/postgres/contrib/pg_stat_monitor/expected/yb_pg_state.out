CREATE EXTENSION pg_stat_monitor;
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

-- https://github.com/yugabyte/yugabyte-db/issues/11801
-- TODO: pg_stat_monitor_reset does not remove this from shared memory and hence this might appear
-- in subsequent pg_stat_monitor tables.
-- SELECT 1;
SELECT 1/0;   -- divide by zero
ERROR:  division by zero
SELECT query, state_code, state FROM pg_stat_monitor ORDER BY query COLLATE "C";
                                      query                                       | state_code |        state
----------------------------------------------------------------------------------+------------+---------------------
 SELECT 1/0;                                                                      |          4 | FINISHED WITH ERROR
 SELECT pg_stat_monitor_reset();                                                  |          3 | FINISHED
 SELECT query, state_code, state FROM pg_stat_monitor ORDER BY query COLLATE "C"; |          2 | ACTIVE
(3 rows)

SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset
-----------------------

(1 row)

DROP EXTENSION pg_stat_monitor;
